### PR TITLE
Add AttendanceCommand and TrainingCommand

### PR DESCRIPTION
**Pull request by jeunhoe**
Hi all. I've done up the attendance and training commands for our application. Please note:

1. Current implementation of the Training command does not save data into storage at the moment. As of now, it only works on runtime memory so opening a new session does not load previous training sessions. Storage has been partially implemented in a separate branch (in this branch) so please do not merge it yet!

2. Attendance command currently displays attendance in the command result box as we have not done up the Main Window for our application. Attendance will be displayed in the Main Window once it is done up.

Please glance through my code if you guys have the time, much appreciated! 🙃

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/TrainingCommand.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        this.indexList = indexList;
+    }
+
+    public TrainingCommand(Date date, List<Index> indexList) {
+        this.date = date;
+        this.indexList = indexList;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```
Comment (hellodommy) [inline src/main/java/seedu/address/logic/commands/TrainingCommand.java:45]: This is a really long method 😯.
Can consider abstracting some parts out for readability after `v1.2`.

File: src/main/java/seedu/address/model/Attendance.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package seedu.address.model.attendance;
```
Comment (hellodommy) [inline src/main/java/seedu/address/model/Attendance.java:1]: Just curious - why isn't `Attendance` in its own `attendance` package anymore?

File: src/main/java/seedu/address/model/training/Training.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package seedu.address.model.training;
 
 import java.util.ArrayList;
+import java.util.Date;
```
Comment (hellodommy) [inline src/main/java/seedu/address/model/training/Training.java:4]: `TrainingDate` class has just been merged 🎉
Can use that now.

File: src/main/java/seedu/address/logic/commands/AttendanceCommand.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        Attendance attendance = model.getAttendance();
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person person = lastShownList.get(index.getZeroBased());
+
+        StringBuilder result = new StringBuilder();
```
Comment (hellodommy) [inline src/main/java/seedu/address/logic/commands/AttendanceCommand.java:54]: Use of `StringBuilder` here makes the insertion of message very clear! 👍🏻
Although appending to a `String` would suffice as well.

**Review by hellodommy — COMMENTED** [review]
Impressive work Jeunhoe! I like how thoroughly implemented this is, as well as the effort on creating the tests.
Looks good to merge 👍🏻

File: src/main/java/seedu/address/logic/commands/AttendanceCommand.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
+package seedu.address.logic.commands;
```
Comment (ChangUo79) [inline src/main/java/seedu/address/logic/commands/AttendanceCommand.java:1]: Good job Jeun Hoe! Just to check with you, will you be implementing the saving of data into json after v1.2?

**Pull request merged**